assessments/projects/cubefs/self-assessment.md

```diff
@@ -0,0 +1,300 @@
+
+* **Performance-Enhancing Security Measures** - 
+
+  * **Goal**: Ensure security measures do not compromise system performance.
+  * **Security Guarantee**: Continuous refinement of security mechanisms (e.g., optimizing writeback cache, addressing performance limitations in encryption) without compromising overall system performance.
+
+* **Key Management and Rotation**:
+
+  * **Goal**: Safeguard shared keys and support regular key rotation for enhanced security.
+  * **Security Guarantee**: Future enhancements to support key rotation, reducing the risk of key compromise and strengthening encryption mechanisms.
+
+### Non-goals
+
+* **Absolute Prevention of All Security Threats** - CubeFS does not claim to completely eliminate all possible security threats or vulnerabilities within the system. While CubeFS implements robust security measures, absolute prevention of all security threats is not feasible due to evolving security landscapes.
+
+* **Full Resistance Against Zero-Day Exploits** - CubeFS does not guarantee complete immunity against zero-day exploits or unknown vulnerabilities. They acknowledge the possibility of unforeseen vulnerabilities emerging.
+
+* **Elimination of All Performance Impact Due to Security Measures** - CubeFS does not aim to implement security measures without any impact on system performance.
+
+* **Providing Certification or Compliance by Default** - CubeFS does not inherently provide certification or compliance with specific security standards by default. Achieving specific certifications or compliance may require additional configurations or procedures.
+
+* **Absolute Guarantee Against Insider Threats** - CubeFS does not promise absolute protection against insider threats or malicious activities from authorized users.
+
+* **Elimination of All Operational Errors** - CubeFS does not aim to eradicate all potential errors or misconfigurations during operational usage.
+
+* **Providing Protection Beyond Defined Use Cases** - CubeFS does not extend security protections to scenarios or use cases not explicitly defined or supported. CubeFS focuses on securing the defined use cases and may not cover security aspects outside these boundaries.
+
+## Self-assessment use
+
+This self-assessment is created to perform an internal analysis of the project's security.  It is not intended to provide a security audit of CubeFS, or function as an independent assessment or attestation of CubeFS's security health. This document serves to provide CubeFS users with an initial understanding of CubeFS's security, where to find existing security documentation, their plans for security, and a general overview of its security practices, both for the development as well as security of CubeFS. This document provides the CNCF TAG-Security with an initial understanding of the project and its security posture.
+
+## Security functions and features

```

Comment:
Considering the project is in cloud storage domain, pointers on data backup, recovery could be useful to include here.